service/src/main/java/bio/terra/workspace/service/workspace/model/CloudPlatform.java

```diff
@@ -6,7 +6,8 @@
 
 public enum CloudPlatform {
   GCP("GCP", ApiCloudPlatform.GCP),
-  AZURE("AZURE", null);
+  AZURE("AZURE", null),
+  MULTI("MULTI", null);
```

Comment:
I agree that this enum is only applicable in reference to a particular cloud. I don't know why references might need to be cloud-bound, but I never quite got a handle on all the use cases a reference supports. If we wanted a general URI reference type, it would be cloud-independent, though I don't know what you'd do with it.

Reply:
I think this is a spot our object model doesn't fit quite right. E.g. we already support references to TDR snapshots - those have the "GCP" cloud platform, but they're not really GCP resources. In retrospect I think `CloudPlatform` should probably have been a ControlledResource property, not a top-level WsmResource property, though changing that now would be a big mess for the API. The only value I can think of for having this at the WsmResource level would be filtering the results of an EnumerateResources call, though we don't have that as a parameter now and I'm not sure any users have asked for it.

References don't (and definitely shouldn't!) require a cloud context.

Reply:
Thanks @jaycarlton and @zloery! I think we can live with this structure more or less how Yu has it.
Yu: I'll leave it to your aesthetic sense to pick a name.

ITFOT, we can move the cloud platform to the controlled resources. I think the API is OK. Saying `/gcp/bucket` and (just for example)`/aws/bucket` seems fine to me to identify the cloud resource.

Reply:
used ANY. Added comment and a TODO to move it to controlled resources.